src/app/form-page/form-page.component.ts

```diff
@@ -0,0 +1,273 @@
+        Validators.pattern("^[a-zA-ZąčęėįšųūžĄČĘĖĮŠŲŪŽ \\.,\\-'\"]+$")
+      ]
+    ],
+    contract: ["", [Validators.required]],
+    contractExplanation: [
+      "",
+      [
+        requiredIfValidator(() => this.applicationForm.get("contract").value),
+        Validators.maxLength(250),
+        Validators.pattern("^[a-zA-ZąčęėįšųūžĄČĘĖĮŠŲŪŽ \\.,\\-'\"]+$")
+      ]
+    ],
+    shift: ["", [Validators.required]],
+    shiftExplanation: [
+      "",
+      [
+        requiredIfValidator(() => this.applicationForm.get("shift").value),
+        Validators.maxLength(250),
+        Validators.pattern("^[a-zA-ZąčęėįšųūžĄČĘĖĮŠŲŪŽ \\.,\\-'\"]+$")
+      ]
+    ],
+    hobbies: [
+      "",
+      [
+        Validators.required,
+        Validators.maxLength(450),
+        Validators.pattern("^[a-zA-ZąčęėįšųūžĄČĘĖĮŠŲŪŽ \\.,\\-'\"]+$")
+      ]
+    ],
+    motivation: [
+      "",
+      [
+        Validators.required,
+        Validators.maxLength(450),
+        Validators.pattern("^[a-zA-ZąčęėįšųūžĄČĘĖĮŠŲŪŽ \\.,\\-'\"]+$")
+      ]
+    ],
+    experience: [
+      "",
+      [
+        Validators.required,
+        Validators.maxLength(450),
+        Validators.pattern("^[a-zA-ZąčęėįšųūžĄČĘĖĮŠŲŪŽ \\.,\\-'\"]+$")
+      ]
+    ],
+    marketing: [
+      "",
+      [
+        Validators.required,
+        Validators.maxLength(250),
+        Validators.pattern("^[a-zA-ZąčęėįšųūžĄČĘĖĮŠŲŪŽ \\.,\\-'\"]+$")
+      ]
+    ]
+  });
+
+  show() {
+    this.showModal = true;
+  }
+
+  hide() {
+    this.showModal = false;
+  }
+
+  ngOnInit(): void {
+    subscribeToValue(this.applicationForm, "contract", "contractExplanation");
+    subscribeToValue(this.applicationForm, "shift", "shiftExplanation");
+    this.$universites = this.formService.fetchSchools();
+    this.applicationForm.controls["establishment"].setValue(0);
+  }
+
+  onSubmit(): void {
+    this.applicationForm.markAllAsTouched();
+    this.tempPerson = null;
+    this.tempAnswerList = [];
+    this.answerPerson = null;
+    let establishmentValue = this.establishment.value;
+    if (establishmentValue === "kita") {
+      establishmentValue = this.establishmentOther.value;
+    }
+    // TODO kas nores gales refactorint
+    this.tempPerson = new Person(
+      this.fname.value,
+      this.lname.value,
+      this.phone.value,
+      this.email.value,
+      establishmentValue
+      // this.establishment.value
+    );
+    this.tempAnswerList.push(new Answer("1", this.contractExplanation.value));
+    this.tempAnswerList.push(new Answer("2", this.shiftExplanation.value));
+    this.tempAnswerList.push(new Answer("3", this.hobbies.value));
+    this.tempAnswerList.push(new Answer("4", this.motivation.value));
+    this.tempAnswerList.push(new Answer("5", this.experience.value));
+    this.tempAnswerList.push(new Answer("6", this.marketing.value));
+    this.answerPerson = new AnswerPerson(this.tempAnswerList, this.tempPerson);
+
+    console.log(JSON.stringify(this.answerPerson));
```

Comment:
console.logs should not be in repository.

Reply:
This is fixed and pushed to frontend-code-fixes.